src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,36 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null || user.getLogin() == null || user.getPassword() == null
+                || user.getAge() == null) {
+            throw new RegistrationDataException("User data cannot be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationDataException("User with this login already exists");
+        }
+        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationDataException("Login must be at least "
+                    + MIN_LOGIN_LENGTH + " characters");
```

Comment:
While checking the login length is good, consider also validating that the login does not contain spaces or special characters if that's a requirement in your system.